src/Microsoft.AspNetCore.Routing.Abstractions/LinkGenerator.cs

```diff
@@ -2,18 +2,21 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System.Collections.Generic;
+using Microsoft.AspNetCore.Http;
 
 namespace Microsoft.AspNetCore.Routing
 {
-    public interface ILinkGenerator
+    public abstract class LinkGenerator
     {
-        bool TryGetLink(
+        public abstract bool TryGetLink(
+            HttpContext httpContext,
```

Comment:
Since this is an optional parameter should this come after other parameter?

Reply:
I am not sure if that would still be clear(if its optional or not) for an end user. Documentation of this API should help.

Reply:
Let's discuss this in during API review though.